tests/test_core.py

```diff
@@ -1859,76 +1870,127 @@ class TestO:
 
         self.assertIsNone(wr_python_object())
 
-    def test_objcinstance_release_owned(self):
-        # Create an object which we own.
-        obj = NSObject.alloc().init()
+    def test_objcinstance_returned_lifecycle(self):
+        """An object is retained when creating an ObjCInstance for it and autoreleased
+        when the ObjCInstance is garbage collected.
+        """
+        with autoreleasepool():
+            # Return an object which we don't own. Using str(uuid) here instead of a
+            # common string ensure that we get have the only reference.
+            obj = NSString.stringWithString(str(uuid.uuid4()))
 
-        # Check that it is marked for release.
-        self.assertTrue(obj._needs_release)
+        # Check that the object is retained when we create the ObjCInstance.
+        self.assertEqual(obj.retainCount(), 1, "object was not retained")
```

Comment:
The assertion doesn't seem to match the comment; should this be:
```suggestion
        self.assertEqual(obj.retainCount(), 1, "object was retained")
```
or does the comment need to be clarified?

Reply:
The message is printed as an error message in case of an assertion failure, so it should be the "opposite" of the inline comment.

Reply:
🤦 Of course.